glean-core/src/lib.rs

```diff
@@ -154,6 +157,41 @@ impl Glean {
             }
         }
     }
+
+    /// Send a ping by name.
+    ///
+    /// The ping content is assembled as soon as possible, but upload is not
+    /// guaranteed to happen immediately, as that depends on the upload
+    /// policies.
+    ///
+    /// If the ping currently contains no content, it will not be sent.
+    ///
+    /// Returns true if a ping was sent, false otherwise.
+    /// Returns an error if collecting or writing the ping to disk failed.
+    pub fn send_ping_by_name(&self, ping_name: &str, log_ping: bool) -> Result<bool> {
+        match self.get_ping_by_name(ping_name) {
+            None => {
+                log::error!("Unknown ping type {}", ping_name);
+                Ok(false)
+            }
+            Some(ping) => self.send_ping(ping, log_ping),
+        }
+    }
+
+    pub fn get_ping_by_name(&self, ping_name: &str) -> Option<&PingType> {
+        self.ping_registry.get(ping_name)
+    }
+
+    pub fn register_ping_type(&mut self, ping: &PingType) {
+        if self.ping_registry.contains_key(&ping.name) {
+            log::error!("Duplicate ping named {}", ping.name)
+        }
+
+        // TODO: This just clones the PingType.  This means we have a copy of the PingType
+        // for the ping registry and another managed by handles on the language-specific
+        // side, but these objects are small and this seems like a simple solution...
```

Comment:
For now they are small. We can optimize this when it becomes a problem.

Reply:
If you agree this is ok, it's ok for me.  I felt this was a little "we couldn't figure out how to disable the borrow checker, so we gave up", and there is probably a way to use handles here.  But the duplication isn't a real problem, other than very small amounts of memory, since they are immutable "data classes" essentially.